### ISSUE TITLE
Add missing SLALIB functions

### ISSUE THREAD

**Issue by cjordan** [Starlink/pal#15]
Currently, I require the following functions to be implemented in `pal`:

- `palRanorm`
- `palGmst`
- `palDcs2c`
- `palDcc2s`
- `palDvn`
- `palDvdv`
- `palDmxv`

In addition, the code base I've inherited uses `slaDranrm`, but as far as I can tell, this is just the same as `slaRanorm`.

As a test, I tried to add my own `palRanorm.c` file into the 0.9.7 version's `Makefile.am`, but attempting then make fails with:
```
root@4c0d7eaedd5f:/automake# make
CDPATH="${ZSH_VERSION+.}:" && cd . && "/automake/t/wrap/aclocal-1.15"
Can't locate /automake/bin/aclocal at /automake/t/wrap/aclocal-1.15 line 29.
Makefile:2472: recipe for target 'aclocal.m4' failed
make: *** [aclocal.m4] Error 2
```

I then tried to compile the Starlink version of `automake`, but this had its own set of problems.

I'm happy to aid converting the FORTRAN code to C, but will need help to make this all work in `pal`.

A (possibly) suitable `palRanorm.c` toward implementing these functions is:
```
/*
*+
*  Name:
*     palRanorm

*  Purpose:
*     Reduce angle to be within [0, 2*pi) (single precision)

*  Language:
*     Starlink ANSI C

*  Type of Module:
*     Library routine

*  Invocation:
*      float palRanorm ( float angle );

*  Arguments:
*     angle = float (Given)
*        Angle to be reduced (radians)

*  Description:
*

*  Authors:
*     TIMJ: Tim Jenness (JAC, Hawaii)
*     PTW: Patrick T. Wallace
*     CHJ: Christopher H. Jordan
*     {enter_new_authors_here}

*  Notes:
*

*  History:
*     2018-06-28 (CHJ):
*        Initial version
*        Adapted from the Fortran SLALIB library.
*     {enter_further_changes_here}

*  Copyright:
*     Copyright (C) 2001 Rutherford Appleton Laboratory
*     Copyright (C) 2012 Science and Technology Facilities Council.
*     All Rights Reserved.

*  Licence:
*     This program is free software; you can redistribute it and/or
*     modify it under the terms of the GNU General Public License as
*     published by the Free Software Foundation; either version 3 of
*     the License, or (at your option) any later version.
*
*     This program is distributed in the hope that it will be
*     useful, but WITHOUT ANY WARRANTY; without even the implied
*     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
*     PURPOSE. See the GNU General Public License for more details.
*
*     You should have received a copy of the GNU General Public License
*     along with this program; if not, write to the Free Software
*     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
*     MA 02110-1301, USA.

*  Bugs:
*     {note_any_bugs_here}
*-
*/

#include <math.h>
#include "pal.h"

float palRanorm ( float angle ) {
  float twoPi = 6.283185307179586476925287;
  float reduced = remainderf(angle, twoPi);
  if (reduced < 0.0) reduced += twoPi;
  return reduced;
}
```

**Comment by cjordan** [comment]
Thanks again for your patience @timj, after doing a little digging, I see that all of my missing functions are indeed already in PAL - sorry about that!

Regarding `palDranorm` - for our astronomy codes, I see three paths forward:

1. I monkey-patch a PAL source file with my implementation above (also utilising `PAL__D2PI`!) whenever we deploy PAL;
2. I edit the single code base I'm currently working on to simply re-cast floats as doubles and back again for use with `palDranrm`.  I would opt for this, except that I'm not sure where other `slaDranorm` function calls are lurking in other codes;
3. PAL includes `palDranorm`.

Personally, I think the choice is yours.  I can understand why you would not want option 3, but if it is an easy change, then it would be appreciated.

**Comment by timj** [comment]
I'm perfectly okay with option 3 -- please make a pull request to add an implementation. The only reason I didn't implement it the first time round was that I didn't expect people to be using it.

**Comment by cjordan** [comment]
Many thanks @timj!

(Merged #16)
